R/zzz.R

```diff
@@ -0,0 +1,29 @@
+.onAttach <- function(libname, pkgname) {
+  message("Loading kinfitr")
+  message("new debug message")
+  # Display opt-out message
+  if (
+      !isTRUE(getOption("kinfitr.no_track")) &&
+        Sys.getenv("KINFITR_NO_TRACK") != "TRUE") {
+    message(
+      "Opt-out of sending tracking information to the KinFitR developers. ",
+      "This information provides an indicator of real world usage crucial for ",
+      "obtaining funding. To disable tracking set ",
+      "options(kinfitr.no_track = TRUE) or the environment variable ",
+      "KINFITR_NO_TRACK to TRUE), to hide this message set ",
+      "options(kinfitr.no_track = FALSE) or the environment variable ",
+      "KINFITR_NO_TRACK to FALSE"
+    )
+  }
+  # # Check if user has opted out
+  if (
+      isTRUE(getOption("kinfitr.no_track")) ||
+        isTRUE(Sys.getenv("KINFITR_NO_TRACK"))) {
+    # do nothing
+    NULL
+  } else {
+    # Send telemetry when package is loaded
+    message("Sending telemetry")
+    send_telemetry(list("kinfitr_usage" = "package_loaded"))
```

Comment:
Maybe I'm being a bit stupid, but what exactly is in the telemetry data? Or is that to be defined? I'd be happy with just an IP address. Is there anything else you usually collect? OS?

Reply:
It's whatever you want to collect, it takes a json like object and posts it. In this instance all it sends is that the package was loaded, the IP address is collected on the POST, de-resolved/anonymized, and the time recorded at the server. You can play around with this using the get_telemetry function or by manually navigating to `http://54.144.240.214/check/kinfitr/` in your browser.